### PR TITLE
OHRM5X-1888: Upgrade oxd dependencies

### PR DESCRIPTION
**Pull request by Super-Chama**
## Checklist

- [x] Test Coverage is 100% for the newly added code
- [x] Storybook stories are added/updated for the changed areas
- [x] Code is linted properly
- [x] Developer testing is done for the affected areas
- [x] Package version updated

### CODE REVIEW

File: components/.eslintrc.js [hunk 1]
```diff
@@ -7,8 +7,7 @@ module.exports = {
     'plugin:vue/vue3-essential',
     'eslint:recommended',
     '@vue/typescript/recommended',
-    '@vue/prettier',
-    '@vue/prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
```
Comment (Super-Chama) [inline components/.eslintrc.js:10]: `@vue/eslint-config-prettier` package remove due to deprecation. `eslint-plugin-vue` is added.

https://cli.vuejs.org/migrations/migrate-from-v4.html#eslint-plugin

File: components/.browserslistrc [hunk 2]
```diff
@@ -1,6 +1,4 @@
-ie >= 11
-edge >= 13
-firefox >= 5
-opera >= 12
-safari >= 5
-chrome >= 56
+> 1%
```
Comment (Super-Chama) [inline components/.browserslistrc:1]: Recommended latest browserlist config https://github.com/vuejs/vue-cli/pull/5233

File: components/jest.config.js [hunk 3]
```diff
@@ -1,7 +1,4 @@
 module.exports = {
   preset: '@vue/cli-plugin-unit-jest/presets/typescript-and-babel',
-  transform: {
-    '^.+\\.vue$': 'vue-jest',
-  },
```
Comment (Super-Chama) [inline components/jest.config.js:5]: `vue-jest` deprecated. `vue3-jest` is installed
https://github.com/vuejs/vue-jest#installation

File: package.json [hunk 4]
```diff
@@ -15,10 +15,11 @@
     "tsc": "yarn workspace oxd-components tsc"
   },
   "engines": {
-    "node": ">=10.19.0",
-    "yarn": ">=1.22.4"
+    "node": ">=16.17.0",
```
Comment (Super-Chama) [inline package.json:18]: Minimum: Node 16 (Active updates) 
Recommended: Node 18 (Current LTS)
https://endoflife.date/nodejs

Yarn 3.3 Latest (version barry)

File: storybook/.storybook/main.js [hunk 5]
```diff
@@ -1,6 +1,9 @@
 const path = require('path');
 
 module.exports = {
+  core: {
+    builder: 'webpack5',
```
Comment (Super-Chama) [inline storybook/.storybook/main.js:5]: Storybook recommended configuration
https://github.com/storybookjs/storybook/tree/main/lib/builder-webpack5#builder-webpack5

File: components/package.json [hunk 6]
```diff
@@ -8,39 +8,37 @@
...
-    "eslint-plugin-vue": "^7.5.0",
-    "prettier": "^1.19.1",
-    "sass": "^1.26.5",
-    "sass-loader": "^8.0.2",
-    "svg-sprite-loader": "^5.2.1",
-    "svg-transform-loader": "^2.0.13",
-    "svgo-loader": "^2.2.1",
-    "typescript": "~4.1.5",
-    "vue-jest": "^5.0.0-0"
+    "@vue/vue3-jest": "^27.0.0-alpha.1",
```
Comment (Super-Chama) [inline components/package.json:31]: Cannot update to `vue-jest 29` based on `jest 29` due to `@vue/cli-plugin-unit-jest` plugin for `vue-cli` is not supported yet. (most likely will not be supported ever as upstream development is now focused on vite)

File: components/package.json [hunk 7]
```diff
@@ -8,39 +8,37 @@
...
+    "@vue/cli-service": "~5.0.8",
+    "@vue/eslint-config-typescript": "^9.1.0",
+    "@vue/test-utils": "^2.0.0",
+    "@vue/vue3-jest": "^27.0.0",
+    "babel-jest": "^27.0.6",
+    "eslint": "^7.32.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-vue": "^8.0.3",
+    "jest": "^27.1.0",
```
Comment (Super-Chama) [inline components/package.json:37]: Cannot update to `vue-jest 29` based on `jest 29` due to `@vue/cli-plugin-unit-jest` plugin for `vue-cli` is not supported yet. (most likely will not be supported ever as upstream development is now focused on vite)

File: components/src/core/components/Input/__tests__/select-input.spec.ts [hunk 8]
```diff
@@ -40,6 +40,10 @@ const options = [
 ];
 
 describe('SelectInput.vue', () => {
+  const div = document.createElement('div');
```
Comment (Super-Chama) [inline components/src/core/components/Input/__tests__/select-input.spec.ts:43]: Due to a bug in jsdom v16 and above, focus is not emitted unless 'attachedTo' value is added to wrapper.  Newer version of vue test utils documentation is missing this. (below exempt is taken from [vue2 docs](https://vue-test-utils.vuejs.org/api/wrapper/#trigger))

refer discussion in https://github.com/vuejs/vue-test-utils/pull/1777

image

File: components/.browserslistrc (hunk 2 above)
Comment (RajithaKumara) [reply components/.browserslistrc:1]: We have to define the minimum supported browser versions.

**Review by RajithaKumara — APPROVED** [review]

File: components/.browserslistrc (hunk 2 above)
Comment (Super-Chama) [reply components/.browserslistrc:1]: >1% is [all versions with >1% of the audience worldwide.](https://browsersl.ist/#q=%3E+1%25)
We can further specify versions, but needs to be updated manually. Let's discuss

From [browserlist tool](https://browsersl.ist)
image

**Pull request merged**